scripts/process-image-manifests

```diff
@@ -0,0 +1,119 @@
+}
+
+trap 'catch $?' EXIT
+
+check_yq() {
+    yq > /dev/null
+}
+
+extract_manifests_from_image() {
+    echo "Creating ${MANIFEST_TMP_DIR} directory"
+    mkdir -p ${MANIFEST_TMP_DIR}
+    echo "Extracting manifests from image '${IMAGE}' to ${MANIFEST_TMP_DIR}"
+    oc image extract $IMAGE --path /manifests/:${MANIFEST_TMP_DIR} --confirm
+    ls -la ${MANIFEST_TMP_DIR}
+}
+
+get_current_csv() {
+    echo "Getting current csv  from existing package.yaml"
+    # The sed entry here will need to be fixed up for each product.
+    # The operator versions do not all follow the same syntax
+    CURRENT_CSV=$(yq r ${MANIFESTS_DIR}/integreatly-${PRODUCT}/${PRODUCT}.package.yaml channels[0].currentCSV | sed -e s/^3scale-operator.v//)
+    echo $CURRENT_CSV
+}
+
+get_new_csv() {
+    IMAGE_MANIFEST_FILE=$(find ${MANIFEST_TMP_DIR} -name "*.package.yaml")
+    DEFAUlT_CHANNEL=$(yq r ${MANIFEST_TMP_DIR}/${PRODUCT_OPERATOR}.package.yaml defaultChannel)
+    NEW_CSV=$(yq r ${MANIFEST_TMP_DIR}/${PRODUCT_OPERATOR}.package.yaml channels.\(name==${DEFAUlT_CHANNEL}\).currentCSV)
+    # The sed entry here will need to be fixed up for each product.
+    # The operator versions do not all follow the same syntax
+    NEW_CSV_SEMVER=$(echo ${NEW_CSV} | sed s/^3scale-operator.v//)
+}
+
+check_versions() {
+    if [ $(ver $NEW_CSV_SEMVER) -eq $(ver $CURRENT_CSV) ]
+    then
+        echo "Versions are the same so EXIT"
+        exit
+    fi
+    if [ $(ver $NEW_CSV_SEMVER) -lt $(ver $CURRENT_CSV) ]
+    then
+        echo "There is a newer version of the CSV present so EXIT"
+        exit
+    fi
+    echo "Newer version found, continuing"
+}
+
+copy_new_csv() {
+    echo "Copying process manifests to ${MANIFESTS_DIR}"
+    # The below line won't work for apicurito as it doesn't add the product name to the semver in it's folder structure
+    NEW_CSV_FOLDER=${PRODUCT}-$NEW_CSV_SEMVER
+    mv ${MANIFEST_TMP_DIR}/$NEW_CSV_SEMVER ${MANIFEST_TMP_DIR}/${NEW_CSV_FOLDER}
+    mv ${MANIFEST_TMP_DIR}/${NEW_CSV_FOLDER} ${MANIFESTS_DIR}/integreatly-${PRODUCT}/
+}
+
+update_new_csv() {
+    echo "updating csv"
+    echo "${MANIFESTS_DIR}/integreatly-${PRODUCT}/${NEW_CSV_FOLDER}/${PRODUCT_OPERATOR}.v$NEW_CSV_SEMVER.clusterserviceversion.yaml"
+    yq d -i ${MANIFESTS_DIR}/integreatly-${PRODUCT}/${NEW_CSV_FOLDER}/${PRODUCT_OPERATOR}.v$NEW_CSV_SEMVER.clusterserviceversion.yaml spec.replaces
+    yq w -i ${MANIFESTS_DIR}/integreatly-${PRODUCT}/${NEW_CSV_FOLDER}/${PRODUCT_OPERATOR}.v$NEW_CSV_SEMVER.clusterserviceversion.yaml spec.install.spec.deployments[0].spec.template.spec.containers[0].env[0].valueFrom.fieldRef.fieldPath metadata.annotations['olm.targetNamespaces']
+}
+
+update_package_yaml() {
+    yq w -i ${MANIFESTS_DIR}/integreatly-${PRODUCT}/${PRODUCT}.package.yaml channels.\(name==integreatly\).currentCSV $NEW_CSV
+}
+
+process_image() {
+   check_yq
+   echo "~~~~~~"
+   echo "Process Image '$IMAGE'"
+   echo "~~~~~~"
+   extract_manifests_from_image
```

Comment:
We might want to consider modifying this at a later stage to allow us specify an image or a directory containing manifests. This would allow others to use this script to copy in CSVs more easily from other non productized operators e.g.. CRO